readme.md

```diff
@@ -581,48 +527,32 @@ const Component = () => {
       >
         <Button />
       </Provider>
-  )
-}
+    )
+  }
   ```
 </details>
 
-## Typing your store and `combine` middleware
+## TypeScript Usage
 
-```tsx
-// You can use `type`
-type BearState = {
-  bears: number
-  increase: (by: number) => void
-}
+Basic typescript usage doesn't require anything special except for writing `create<State>()(...)` instead of `create(...)`...
+
+```ts
+import create from "zustand"
+import { devtools, persist } from "zustand/middleware"
 
-// Or `interface`
 interface BearState {
   bears: number
   increase: (by: number) => void
 }
 
-// And it is going to work for both
-const useStore = create<BearState>(set => ({
+const useStore = create<BearState>()(devtools(persist((set) => ({
   bears: 0,
-  increase: (by) => set(state => ({ bears: state.bears + by })),
-}))
+  increase: (by) => set((state) => ({ bears: state.bears + by })),
+}))))
 ```
 
-Or, use `combine` and let tsc infer types. This merges two states shallowly.
-
-```tsx
-import { combine } from 'zustand/middleware'
-
-const useStore = create(
-  combine(
-    { bears: 0 },
-    (set) => ({ increase: (by: number) => set((state) => ({ bears: state.bears + by })) })
-  ),
-)
-```
+A more complete TypeScript guide is [here](https://github.com/pmndrs/zustand/blob/main/docs/typescript.md).
```

Comment:
Can we make it relative?
```suggestion
A more complete TypeScript guide is [here](./docs/typescript.md).
```

Reply:
I'm not sure if it'll work, probably it won't.

Reply:
I'm sure it works in github, but not in npmjs.
It was "Not Found" for me when browsing the commit instead of `main`.